tico/experimental/quantization/ptq/wrappers/quant_module_base.py

```diff
@@ -0,0 +1,153 @@
+from tico.experimental.quantization.ptq.observers.base import ObserverBase
+from tico.experimental.quantization.ptq.quant_config import QuantConfig
+
+
+class QuantModuleBase(nn.Module, ABC):
+    """
+    Abstract parent for EVERY wrapper.
+
+    Responsibilities
+    ----------------
+    • Own *one* Mode enum (`NO_QUANT / CALIB / QUANT`)
+    • Own a QuantConfig describing default / per-observer dtypes
+    • Expose a canonical lifecycle:
+          enable_calibration()
+          freeze_qparams()
+    • Provide helper `_fq(x, observer)` (“fake-quant or collect”) so
+      subclasses write arithmetic code without boilerplate.
+    """
+
+    def __init__(
+        self, qcfg: Optional[QuantConfig] = None, *, fp_name: Optional[str] = None
+    ) -> None:
+        super().__init__()
+        self.qcfg = qcfg or QuantConfig()
+        self._mode: Mode = Mode.NO_QUANT  # default state
+        self.fp_name = fp_name
+
+    def _child_quant_modules(self):
+        """Yield direct children that are QuantModuleBase."""
+        for m in self.children():
+            if isinstance(m, QuantModuleBase):
+                yield m
+
+    def enable_calibration(self) -> None:
+        self._mode = Mode.CALIB
+        for obs in self._all_observers():
+            obs.enabled = True
+            obs.reset()
+
+        # propagate to children
+        for child in self._child_quant_modules():
+            child.enable_calibration()
+
+    def freeze_qparams(self) -> None:
+        self._mode = Mode.QUANT
+        for obs in self._all_observers():
+            obs.enabled = False
+            obs.compute_qparams()
+
+        # propagate to children
+        for child in self._child_quant_modules():
+            child.freeze_qparams()
+
+    def _fq(self, x, obs: ObserverBase):
+        """Fake-quant or collect."""
+        if self._mode is Mode.CALIB:
+            obs.collect(x.detach())
+            return x
+        if self._mode is Mode.QUANT:
+            return obs.fake_quant(x)
+        return x  # NO_QUANT
+
+    @abstractmethod
+    def _all_observers(self) -> Iterable[ObserverBase]:
+        """Return every observer owned by this module."""
+        ...
+
+    def named_observers(self) -> Iterable[Tuple[str, ObserverBase]]:
+        for obs in self._all_observers():
+            yield obs.name, obs
+
+    def get_observer(self, name: str) -> Optional[ObserverBase]:
+        for obs in self._all_observers():
+            if obs.name == name:
+                return obs
+        return None
+
+    def _make_obs(
+        self,
+        name: str,
+        **default_kwargs,
+    ) -> ObserverBase:
+        """
+        Instantiate an observer named *name*.
+
+        Precedence (3-tier) for keys:
+           • observer:  user > wrapper-default > QuantConfig.default_observer
+           • dtype:     user > wrapper-default > QuantConfig.default_dtype
+           • qscheme:   user  > wrapper-default > QuantConfig.default_qscheme
```

Comment:
```suggestion
           • qscheme:   user > wrapper-default > QuantConfig.default_qscheme
```
Just FYI, there is a blank still remaining
I'm not requesting you to change it, but just in case you forgot it... 😓